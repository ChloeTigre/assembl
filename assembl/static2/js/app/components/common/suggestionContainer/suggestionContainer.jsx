// @flow
import React from 'react';

export type Props = {
  /** List of suggestions fetched from the semantic analysis engine (IBM Watson) */
  suggestionList: Array<string>
};

const SuggestionContainer = ({ suggestionList }: Props) => (
  <div className="suggestion-container">
    <div className="title">Suggestions de mots-cl√©s :</div>
    <div className="suggestion-list">
      {suggestionList.map((suggestion, index) => <span key={`suggestion-${index}`}>{suggestion}</span>)}
    </div>
  </div>
);

export default SuggestionContainer;