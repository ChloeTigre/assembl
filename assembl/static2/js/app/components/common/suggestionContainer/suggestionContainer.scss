.suggestion-container {
  border-top: solid 1px $grey;
  padding-top: 5px;
  padding-bottom: 5px;

  .title {
    font-size: 1.1rem;
    letter-spacing: 1.2px;
  }

  .suggestion-list {
    span {
      display: inline-block;
      font-size: 1rem;
      letter-spacing: normal;
      margin-right: 15px;
    }
  }
}
