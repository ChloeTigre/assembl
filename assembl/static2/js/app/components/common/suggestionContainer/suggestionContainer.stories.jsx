// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
/* eslint-enable */

import SuggestionContainer from './suggestionContainer';

storiesOf('Tag On Post|SuggestionContainer', module)
  .addDecorator(withInfo)
  .add('default', () => <SuggestionContainer />);