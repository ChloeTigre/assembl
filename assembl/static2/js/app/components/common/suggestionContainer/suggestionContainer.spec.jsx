// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import initStoryshots from '@storybook/addon-storyshots';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.3';
/* eslint-enable */

import SuggestionContainer from './suggestionContainer';
import type { Props } from './suggestionContainer';
import { defaultProps } from './suggestionContainer.stories';

initStoryshots({
  storyKindRegex: /^Tag\s{1}On\s{1}Post\|SuggestionContainer$/
});

configure({ adapter: new Adapter() });

describe('<SuggestionContainer /> - with shallow', () => {
  let wrapper;
  let suggestionList: Props;

  beforeEach(() => {
    suggestionList = { ...defaultProps };
    wrapper = shallow(<SuggestionContainer {...suggestionList} />);
  });

  it('should render a title', () => {
    expect(wrapper.find('div [className="title"]')).toHaveLength(1);
  });

  it('should render a list of 5 suggestions', () => {
    expect(wrapper.find('div [className="suggestion-list"]')).toHaveLength(1);
    expect(wrapper.find('span')).toHaveLength(5);
  });
});