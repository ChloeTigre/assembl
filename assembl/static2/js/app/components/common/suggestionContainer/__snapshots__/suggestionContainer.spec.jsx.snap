// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Tag On Post|SuggestionContainer default 1`] = `
<div
  className="suggestion-container"
>
  <div
    className="title"
  >
    Suggested keywords:
  </div>
  <div
    className="suggestion-list"
  >
    <span>
      Investissement
    </span>
    <span>
      Mesures
    </span>
    <span>
      Inclusive
    </span>
    <span>
      Application
    </span>
    <span>
      Faisabilité
    </span>
  </div>
</div>
`;

exports[`Storyshots Tag On Post|SuggestionContainer playground 1`] = `
<div
  className="suggestion-container"
>
  <div
    className="title"
  >
    This is a custom title
  </div>
  <div
    className="suggestion-list"
  >
    <span>
      Investissement
    </span>
    <span>
      Mesures
    </span>
    <span>
      Inclusive
    </span>
    <span>
      Application
    </span>
    <span>
      Faisabilité
    </span>
  </div>
</div>
`;
